server/src/operators/chunk_operator.rs

```diff
@@ -2184,6 +2184,53 @@ pub async fn create_chunk_metadata(
 
     let mut chunk_metadatas = vec![];
 
+    let mut group_tracking_ids_to_group_ids: HashMap<String, uuid::Uuid> = HashMap::new();
+
+    let all_group_tracking_ids: Vec<String> = chunks
+        .iter()
+        .flat_map(|chunk| chunk.group_tracking_ids.clone().unwrap_or_default())
+        .unique()
+        .collect();
+
+    get_group_ids_from_tracking_ids_query(
+        all_group_tracking_ids.clone(),
+        dataset_uuid,
+        pool.clone(),
+    )
+    .await?
+    .iter()
+    .for_each(|(group_id, tracking_id)| {
+        if let Some(tracking_id) = tracking_id {
+            group_tracking_ids_to_group_ids.insert(tracking_id.clone(), *group_id);
+        }
+    });
+
+    let new_groups: Vec<ChunkGroup> = all_group_tracking_ids
+        .iter()
+        .filter_map(|group_tracking_id| {
+            if !group_tracking_ids_to_group_ids.contains_key(group_tracking_id) {
+                Some(ChunkGroup::from_details(
+                    Some(group_tracking_id.clone()),
+                    None,
+                    dataset_uuid,
+                    Some(group_tracking_id.to_string()),
+                    None,
+                    None,
+                ))
+            } else {
+                None
+            }
+        })
+        .collect();
+
+    let created_groups = create_groups_query(new_groups, true, pool.clone()).await?;
```

Comment:
Also setting upsert_by_group_tracking_id to `true` just in case of the race condition that 2 servers are trying to create the group at the same time. We want them to agree on the same id, onConflictDoNothing will not return a group_id at all which might cause some issues.